cmd/update-controller/main.go

```diff
@@ -0,0 +1,147 @@
+package main
+
+import (
+	"errors"
+	"fmt"
+	"strings"
+	"time"
+
+	"github.com/coreos/bootkube/pkg/cluster"
+	"github.com/golang/glog"
+
+	"k8s.io/kubernetes/pkg/api"
+	"k8s.io/kubernetes/pkg/api/v1"
+	"k8s.io/kubernetes/pkg/client/cache"
+	clientset "k8s.io/kubernetes/pkg/client/clientset_generated/release_1_3"
+	"k8s.io/kubernetes/pkg/client/restclient"
+	"k8s.io/kubernetes/pkg/controller/framework"
+	"k8s.io/kubernetes/pkg/fields"
+	"k8s.io/kubernetes/pkg/runtime"
+	"k8s.io/kubernetes/pkg/util/wait"
+	"k8s.io/kubernetes/pkg/watch"
+)
+
+type versionPayload struct {
+	newVersion *cluster.ContainerImage
+	configMap  *v1.ConfigMap
+}
+
+func main() {
```

Comment:
This component should be leader-elected. Can pretty much just pull same functionality from scheduler or controller-manager mains.